### PR TITLE
Allow nested "match" update syntax (with arrays)

### PR DESCRIPTION
**Pull request by jonmv**
@vekterli

### CODE REVIEW

File: document/src/test/java/com/yahoo/document/json/JsonReaderTestCase.java [hunk 1]
```diff
@@ -410,11 +420,20 @@ public void testUpdateWeighted() throws IOException {
...
+                                                   "        'element': 'person',",
+                                                   "        'increment': 13 }}}}"));
+
+        DocumentUpdate otherDoc = parseUpdate(inputJson("{ 'update': 'id:unittest:testset::whee',",
+                                                        "  'fields': {",
+                                                        "    'actualset': {",
+                                                        "      'match': {",
+                                                        "        'increment': 13,",
+                                                        "        'element': 'person' }}}}"));
+
```
Comment (jonmv) [inline document/src/test/java/com/yahoo/document/json/JsonReaderTestCase.java:435]: This fails now, so don't merge :)

**Comment by jonmv** [comment]
Just for discussions, so far! I'll make some actual changes, too, and I guess the question is whether preserving the current behaviour is desirable.

**Comment by jonmv** [comment]
Feel free to review now, @vekterli

File: document/src/test/java/com/yahoo/document/json/JsonReaderTestCase.java [hunk 2]
```diff
@@ -719,15 +885,16 @@ public void testMapStringToArrayOfInt() throws IOException {
...
-        DocumentType docType = r.readDocumentType(parseInfo.documentId);
-        DocumentPut put = new DocumentPut(new Document(docType, parseInfo.documentId));
-        new VespaJsonDocumentReader(false).readPut(parseInfo.fieldsBuffer, put);
-        Document doc = put.getDocument();
+        Document doc = docFromJson("""
+                                   {
+                                     "put": "id:unittest:testMapStringToArrayOfInt::whee",
+                                     "fields": {
+                                       "actualMapStringToArrayOfInt": {
+                                         "bamse": [1, 2, 3]
```
Comment (vekterli) [inline document/src/test/java/com/yahoo/document/json/JsonReaderTestCase.java:893]: This previously appears to have tested a (presumed legacy) update syntax with explicit `"key": foo, "value": bar` mappings. With these changes it's identical to `testMapStringToArrayOfInt()`.

Consider either testing the explicit KV syntax (if we want to test it) or removing this test entirely (if we don't).

**Review by vekterli — DISMISSED** [review]
:+1:

File: document/src/test/java/com/yahoo/document/json/JsonReaderTestCase.java (hunk 2 above)
Comment (jonmv) [reply document/src/test/java/com/yahoo/document/json/JsonReaderTestCase.java:893]: Ah, my mistake. I tried to verify all the AI generated JSON, but evidently I failed here.

**Review by vekterli — APPROVED** [review]

**Pull request merged**